src/libs/ajax/analysis-providers/AnalysisProvider.ts

```diff
@@ -0,0 +1,65 @@
+import { Ajax } from 'src/libs/ajax'
+import { GoogleWorkspaceInfo, isGoogleWorkspaceInfo, WorkspaceInfo } from 'src/libs/workspace-utils'
+import { AnalysisFile } from 'src/pages/workspaces/workspace/analysis/useAnalysisFiles'
+import { AbsolutePath, getExtension, getFileName, stripExtension } from 'src/pages/workspaces/workspace/analysis/utils/file-utils'
+import { getToolLabelFromFileExtension, ToolLabel } from 'src/pages/workspaces/workspace/analysis/utils/tool-utils'
+
+
+export interface AnalysisProviderContract {
+  listAnalyses: (workspaceInfo: WorkspaceInfo, signal?: AbortSignal) => Promise<AnalysisFile[]>
+  copyAnalysis: (
+      sourceWorkspace: WorkspaceInfo,
+      printName: string,
+      toolLabel: ToolLabel,
+      targetWorkspace: WorkspaceInfo,
+      newName: string,
+      signal?: AbortSignal
+  ) => Promise<void>
+  createAnalysis: (workspaceInfo: WorkspaceInfo, fullAnalysisName: string, toolLabel: ToolLabel, contents: any, signal?: AbortSignal) => Promise<void>
+  deleteAnalysis: (workspaceInfo: WorkspaceInfo, path: AbsolutePath, signal?: AbortSignal) => Promise<void>
+}
+
+export const AnalysisProvider: AnalysisProviderContract = {
+  listAnalyses: async (workspaceInfo: WorkspaceInfo, signal?: AbortSignal): Promise<AnalysisFile[]> => {
+    const selectedAnalyses: AnalysisFile[] = isGoogleWorkspaceInfo(workspaceInfo) ?
+      await Ajax(signal).Buckets.listAnalyses(workspaceInfo.googleProject, workspaceInfo.bucketName) :
+      // TODO: cleanup once TS is merged in for AzureStorage module
+      (await Ajax(signal).AzureStorage.listNotebooks(workspaceInfo.workspaceId) as any)
+    return selectedAnalyses
+  },
+  copyAnalysis: async (
+    sourceWorkspace: WorkspaceInfo,
+    printName: string,
+    toolLabel: ToolLabel,
+    targetWorkspace: WorkspaceInfo,
+    newName: string,
+    signal?: AbortSignal
+  ): Promise<void> => {
+    if (isGoogleWorkspaceInfo(sourceWorkspace)) {
+      await Ajax(signal)
+        .Buckets
+        .analysis(sourceWorkspace.googleProject, sourceWorkspace.bucketName, printName, toolLabel)
+        // assumes GCP to GCP copy
+        .copy(`${newName}.${getExtension(printName)}`, (targetWorkspace as GoogleWorkspaceInfo).bucketName, false)
+    } else {
+      await Ajax(signal).AzureStorage
+        .blob(sourceWorkspace.workspaceId, printName)
+        .copy(stripExtension(newName), targetWorkspace.workspaceId)
+    }
+  },
+  createAnalysis: async (workspaceInfo: WorkspaceInfo, fullAnalysisName: string, toolLabel: ToolLabel, contents: any, signal?: AbortSignal): Promise<void> => {
```

Comment:
are we getting unit test coverage on create and delete functions here thru useAnalysisFiles unit tests?  Ideal would be to have unit tests here... but at a minimum, add a comment to the unit test file to state that these two functions are getting coverage and where.